content/hi/microservices.md

```diff
@@ -0,0 +1,18 @@
+---
+title: माइक्रोसर्विसेज(Microservices) 
+status: Completed
+category: संकल्पना
+tags : ["आर्किटेक्चर","मौलिक"]
+---
+
+## यह क्या है
+
+माइक्रोसर्विसेज अनुप्रयोग विकास के लिए एक आधुनिक दृष्टिकोण है जो क्लाउड नेटिव प्रौद्योगिकियों का लाभ उठाता है। जबकि आधुनिक एप्लिकेशन, जैसे नेटफ्लिक्स, एक ही ऐप प्रतीत होते हैं, वे वास्तव में छोटी सेवाओं का एक संग्रह हैं - सभी एक साथ मिलकर काम कर रहे हैं। उदाहरण के लिए, एक एकल पृष्ठ जो आपको वीडियो तक पहुंचने, खोजने और पूर्वावलोकन करने की अनुमति देता है, संभवतः छोटी सेवाओं द्वारा संचालित होता है जो प्रत्येक इसके एक पहलू को संभालती हैं (जैसे खोज, प्रमाणीकरण और आपके ब्राउज़र में पूर्वावलोकन चलाना)। संक्षेप में, माइक्रोसर्विसेज एक एप्लिकेशन आर्किटेक्चर पैटर्न को संदर्भित करता है जो आमतौर पर अखंड अनुप्रयोगों के विपरीत होता है ।
+
+## समस्या 
+
+माइक्रोसर्विसेज अखंड अनुप्रयोगों द्वारा उत्पन्न चुनौतियों का जवाब है। आम तौर पर, किसी एप्लिकेशन के विभिन्न हिस्सों को अलग-अलग स्केल करने की आवश्यकता होगी । एक ऑनलाइन स्टोर, उदाहरण के लिए, चेकआउट की तुलना में अधिक उत्पाद दृश्य होने जा रहे हैं। इसका मतलब है कि आपको चेकआउट की तुलना में चल रही उत्पाद दृश्य कार्यक्षमता की अधिक प्रतियों की आवश्यकता होगी। एक अखंड अनुप्रयोग में, तर्क के उन अंशों को अलग से तैनात नहीं किया जा सकता है। यदि आप उत्पाद की कार्यक्षमता को अलग-अलग माप नहीं सकते हैं, तो आपको पूरे ऐप को अन्य सभी घटकों के साथ डुप्लिकेट करना होगा जिनकी आपको आवश्यकता नहीं है - संसाधनों का अक्षम उपयोग। अखंड अनुप्रयोग भी डेवलपर्स के लिए डिज़ाइन की कमियों के आगे झुकना आसान बनाते हैं। क्योंकि सभी कोड एक ही स्थान पर हैं, इसलिए उस कोड को कसकर युग्मित करना आसान होता हैऔर चिंताओं को अलग करने के सिद्धांत को लागू करना कठिन है। मोनोलिथ्स को अक्सर डेवलपर्स को उत्पादक होने से पहले पूरे कोडबेस को समझने की आवश्यकता होती है।
```

Comment:
tiny nit here. `हैऔर` should be two separate words.

--
last sentence can be better framed as `मोनोलिथ्स की अक्सर आवश्यकता होती है कि डेवलपर्स उत्पादक होने से पहले पूरे कोडबेस को समझें । `